contracts/lottery/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+#![no_std]
+
+use rand::rngs::SmallRng;
+use rand::{Rng, SeedableRng};
+
+use soroban_sdk::{
+    contracterror, contractimpl, contracttype, token, Address, Env, Map, Symbol, Vec,
+};
+
+#[derive(Clone, Copy)]
+#[contracttype]
+enum DataKey {
+    Admin = 1,
+    Candidates = 2,
+    WinnerCount = 3,
+    TicketPrice = 4,
+    Token = 5,
+    AlreadyPlayed = 6,
+}
+
+#[contracterror]
+#[derive(Clone, Debug, Copy, Eq, PartialEq, PartialOrd, Ord)]
+#[repr(u32)]
+pub enum Error {
+    InsufficientFunds = 1,
+    AlreadyPlayed = 2,
+    MinParticipantsNotSatisfied = 3,
+}
+
+pub struct LotteryContract;
+
+#[contractimpl]
+impl LotteryContract {
+    pub fn initialize(
+        env: Env,
+        admin: Address,
+        token: Address,
+        winners_count: u32,
+        ticket_price: i128,
+    ) {
+        admin.require_auth();
+        let storage = env.storage();
+        storage.set(&DataKey::Admin, &admin);
+        storage.set(&DataKey::Token, &token);
+        storage.set(&DataKey::WinnerCount, &winners_count);
+        storage.set(&DataKey::TicketPrice, &ticket_price);
+        storage.set(&DataKey::Candidates, &Vec::<Address>::new(&env));
+        storage.set(&DataKey::AlreadyPlayed, &false);
```

Comment:
Probably in the wizard we would remove the init parameters, and "hardcode" them by substitution.

Reply:
These would be passed in invoke

Reply:
Yes, they already are, this comment is from 3 weeks ago. I will resolve it to not cause confusion.